client/my-sites/checkout/checkout-thank-you/index.jsx

```diff
@@ -438,6 +447,15 @@ export class CheckoutThankYou extends React.Component {
 				);
 			}
 
+			// This is for the DIFM upsell A/B test. Check https://wp.me/pcbrnV-Y3.
+			recordTracksEvent( 'calypso_eligible_difm_upsell' );
+			if (
+				shouldShowDifmUpsell &&
+				! previousRoute.includes( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }` )
+			) {
+				page( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }?isEcommerce=1` );
+			}
```

Comment:
I think it is more accurate to split this?
```suggestion
			// This is for the DIFM upsell A/B test. Check https://wp.me/pcbrnV-Y3.
			if ( ! previousRoute.includes( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }` ) ) {
				recordTracksEvent( 'calypso_eligible_difm_upsell' );

				if (  shouldShowDifmUpsell ) {
        		   	page( `/checkout/${ selectedSiteSlug }/offer-difm/${ receiptId }?isEcommerce=1` );
		    	}
			}
```

Reply:
I think it's okay to have the `recordTracksEvent()`  call before the `if` condition. The `previousRoute` check is only for those users who've already seen the offer page and we don't want to show it to them a second time. So, they've already entered the experiment at that point.

Reply:
Yeah that makes sense to me :)